scipy/linalg/tests/test_decomp_ldl.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import division, print_function, absolute_import
+
+from numpy.testing import assert_array_almost_equal, assert_allclose, assert_
+from numpy import (array, eye, zeros, empty_like, empty, tril_indices_from,
+                   tril, triu_indices_from, spacing)
+from numpy.random import rand, randint, seed
+from scipy.linalg import ldl
+from pytest import raises as assert_raises, warns
+from numpy import ComplexWarning
+
+
+def test_args():
+    A = eye(3)
+    # Nonsquare array
+    assert_raises(ValueError, ldl, A[:, :2])
+    # Complex matrix with imaginary diagonal entries with "hermitian=True"
+    with warns(ComplexWarning):
+        ldl(A*1j)
+
+
+def test_empty_array():
+    a = empty((0, 0), dtype=complex)
+    l, d, p = ldl(empty((0, 0)))
+    assert_array_almost_equal(l, empty_like(a))
+    assert_array_almost_equal(d, empty_like(a))
+    assert_array_almost_equal(p, array([], dtype=int))
+
+
+def test_simple():
+    a = array([[-0.39-0.71j, 5.14-0.64j, -7.86-2.96j, 3.80+0.92j],
```

Comment:
Reading this through again: add a comment how were these arrays generated.

Reply:
Also, there does not appear to be clear reason to use predefined arrays, since they are not ground truth values. So maybe switch to random arrays?

Also, different data types are not tested. This would be simpler with random inputs.

Reply:
I compiled it from other examples I've found around just to compare if it is working properly and did the random part with permutations but indeed I should add a proper random suite testing all dtypes and sizes. I'll add them.